src/XperienceCommunity.Localization.Base/Localizer/XperienceStringLocalizerBase.cs

```diff
@@ -0,0 +1,182 @@
+                inner join CMS_ContentLanguage on ContentLanguageCultureFormat = KeyValue or KeyValue like ContentLanguageName+'%'
+                where KeyName = 'CMSDefaultCultureCode'";
+                return ConnectionHelper.ExecuteQuery(query, [], QueryTypeEnum.SQLQuery).Tables[0].Rows.Cast<DataRow>()
+                .ToDictionary(key => (int)key["WebsiteChannelID"], value => new LanguageSummary((string)value["ContentLanguageName"], (string)value["ContentLanguageCultureFormat"]));
+            }, new CacheSettings(1440, "Localization_WebsiteIdToLanguageInfoDictionary"));
+
+        internal record LanguageSummary(string LanguageName, string CultureName);
+
+        internal string SiteVisitorDefaultCulture
+        {
+            get
+            {
+                var dictionary = WebsiteIdToLanguageInfoDictionary();
+                if (dictionary.TryGetValue(websiteChannelContext.WebsiteChannelID, out var summary))
+                {
+                    return summary.CultureName;
+                }
+                else
+                {
+                    return dictionary.TryGetValue(-1, out var defaultSummary) ? defaultSummary.CultureName : "en-US";
+                }
+            }
+        }
+
+        internal string CMSDefaultCulture => WebsiteIdToLanguageInfoDictionary().TryGetValue(-1, out var defaultSummary) ? defaultSummary.CultureName : "en-US";
+
+        private Dictionary<string, string> GetDictionary(string cultureName)
+        {
+            // Now load up dictionary
+            var langChain = GetLanguageChain(cultureName);
+            if (langChain.Count == 0)
+            {
+                return [];
+            }
+            return progressiveCache.Load(cs =>
+            {
+                if (cs.Cached)
+                {
+                    cs.CacheDependency = CacheHelper.GetCacheDependency([
+                        $"{LocalizationKeyInfo.OBJECT_TYPE}|all",
+                        $"{LocalizationTranslationItemInfo.OBJECT_TYPE}|all"
+                    ]);
+                }
+                // This logic will build out the priority listing for the language fallbacks, picking the most accurate match
+                string orderBySql = "order by ";
+                string ends = "";
+                for (int i = 0; i < langChain.Count; i++)
+                {
+#pragma warning disable S1643 // Strings should not be concatenated using '+' in a loop
+                    orderBySql += $" case when ContentLanguageCultureFormat = '{SqlHelper.EscapeQuotes(langChain[i])}' then {i} else ";
+                    ends += " end ";
+#pragma warning restore S1643 // Strings should not be concatenated using '+' in a loop
+                }
+                orderBySql += $" {langChain.Count} {ends}";
+
+                var results = ConnectionHelper.ExecuteQuery(
+                    $@"select LocalizationKeyItemName as StringKey, LocalizationTranslationItemText as TranslationText from (
```

Comment:
Can you extract this SQL query to a constant?

Reply:
Converted to constant and used string.format to inject the dynamic values.